spec/util/extensions/miq-module_spec.rb

```diff
@@ -41,6 +41,33 @@
       expect(test_module.default).to eq(value)
     end
 
+    it 'clears the cache even if the block raises an exception' do
+      test_class.define_singleton_method(:call_count)  { @call_count ||= 0 }
+      test_class.define_singleton_method(:call_count=) { |val| @call_count = val }
+      test_class.cache_with_timeout(:flapping_method) do
+        begin
+          if test_class.call_count.even?
+            test_class.call_count
+          else
+            raise "OOPS"
+          end
+        ensure
+          test_class.call_count += 1
+        end
+      end
+
+      expect(test_class.flapping_method(true)).to eq(0) # initializes cache to 0
+      expect(test_class.flapping_method).to eq(0)       # returns from cache
+
+      test_class.flapping_method(true) rescue nil       # blows up, clears cache for next request
+      expect(test_class.flapping_method).to eq(2)       # sets new value since cache is cleared
```

Comment:
> In the other PR you mentioned a test for 5, raise, raise, etc. Do you think you still need that?

@Fryguy No.  This expectation shows that we are actually setting a new value and returning it, NOT the previously cached value, 0.  So, I think this test encompasses the desired test cases.